driver/registry.go

```diff
@@ -204,6 +206,23 @@ func ReplaceTracer(f func(*otelx.Tracer) *otelx.Tracer) RegistryOption {
 	}
 }
 
+type NewStrategy func(deps any) any
```

Comment:
WIP?

Reply:
The strategies are all just empty interfaces that are checked at runtime. So this code is as prod as it gets 😎 .